info.limpet.rcp/src/info/limpet/rcp/wizards/SampleProjectWizardPage.java

```diff
@@ -0,0 +1,143 @@
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IWorkspaceRoot;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.FileLocator;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.wizard.WizardPage;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.ModifyEvent;
+import org.eclipse.swt.events.ModifyListener;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.ui.actions.CopyFilesAndFoldersOperation;
+import org.osgi.framework.Bundle;
+
+import info.limpet.rcp.Activator;
+
+public class SampleProjectWizardPage extends WizardPage
+{
+
+	private String projectName;
+
+	protected SampleProjectWizardPage()
+	{
+		super("info.limpet.rcp.wizard.sampleProjectWizardPage");
+		setTitle("Create Sample Project");
+		setDescription("Create Sample Project");
+	}
+
+	@Override
+	public void createControl(Composite parent)
+	{
+		Composite composite = new Composite(parent, SWT.NONE);
+		composite.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true));
+		composite.setLayout(new GridLayout(2, false));
+		new Label(composite, SWT.LEFT).setText("Project name:");
+		final Text nameText = new Text(composite, SWT.SINGLE | SWT.BORDER);
+		projectName = "SampleProject";
+		IProject project = ResourcesPlugin.getWorkspace().getRoot()
+				.getProject(projectName);
+		if (project != null && project.exists())
+		{
+			projectName = "";
+		}
+		nameText.setText(projectName);
+		nameText.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false));
+		nameText.addModifyListener(new ModifyListener()
+		{
+			public void modifyText(ModifyEvent event)
+			{
+				projectName = nameText.getText();
+				validate();
+			}
+		});
+		setControl(composite);
+		setPageComplete(false);
+		validate();
+
+	}
+
+	private void validate()
+	{
+		setErrorMessage(null);
+		setPageComplete(true);
+		if (projectName == null || projectName.isEmpty())
+		{
+			setErrorMessage("Project name is required");
+			setPageComplete(false);
+		}
+		else
+		{
+			IProject project = ResourcesPlugin.getWorkspace().getRoot()
+					.getProject(projectName);
+			if (project != null && project.exists())
+			{
+				setErrorMessage("The '" + projectName + "' exists.");
+				setPageComplete(false);
+			}
+		}
+	}
+
+	public boolean performFinish()
+	{
+		try
+		{
+			IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+			IProject project = root.getProject(projectName);
+			project.create(null);
+			project.open(null);
+			Bundle bundle = Platform.getBundle("info.limpet.sample_data");
+			if (bundle == null)
+			{
+				return true;
```

Comment:
we should be logging an error if this fails - to indicate the bundle can't be found